src/commands/mod.rs

```diff
@@ -289,6 +289,58 @@ impl DaemonControl {
         GetAddressResult::new(address)
     }
 
+    /// list addresses
+    pub fn list_addresses(
+        &self,
+        start_index: Option<u32>,
+        count: Option<u32>,
+    ) -> ListAddressesResult {
+        let mut db_conn = self.db.connection();
+        let receive_index: u32 = db_conn.receive_index().into();
+        let change_index: u32 = db_conn.change_index().into();
+
+        let start_index = start_index.unwrap_or(0);
+
+        let index = if let Some(c) = count {
+            (start_index + c).saturating_sub(1)
+        } else {
+            let max_index = receive_index.max(change_index);
+            if start_index >= max_index {
+                start_index
+            } else {
+                max_index.saturating_sub(1)
+            }
+        };
+
+        let addresses: Vec<AddressInfo> = (start_index..=index)
+            .map(|index| {
+                let child = bip32::ChildNumber::from_normal_idx(index)
+                    .expect("Failed to convert index to ChildNumber");
```

Comment:
Could we error instead of crashing if a value greater than `2**31 - 1` is passed for count? It's an insane argument to give to the command, but it's nicer to never crash on insane inputs.

Let's add an `InvalidAddressCount` variant to the `CommandError` enum and use it here and above?